hypervisor/arch/x86/ept.c

```diff
@@ -115,46 +115,44 @@ int32_t ept_violation_vmexit_handler(struct acrn_vcpu *vcpu)
 	ret = decode_instruction(vcpu);
 	if (ret > 0) {
 		mmio_req->size = (uint64_t)ret;
-	} else if (ret == -EFAULT) {
-		pr_info("page fault happen during decode_instruction");
-		status = 0;
-		goto out;
 	} else {
-		goto out;
+		if (ret == -EFAULT) {
+			pr_info("page fault happen during decode_instruction");
+			status = 0;
+		}
 	}
 
-
 	/*
 	 * For MMIO write, ask DM to run MMIO emulation after
 	 * instruction emulation. For MMIO read, ask DM to run MMIO
 	 * emulation at first.
 	 */
 
 	/* Determine value being written. */
-	if (mmio_req->direction == REQUEST_WRITE) {
-		status = emulate_instruction(vcpu);
-		if (status != 0) {
-			goto out;
+	if (ret > 0) {
+		if (mmio_req->direction == REQUEST_WRITE) {
```

Comment:
move the following code right after Line 117?

Reply:
Done